mvebu: add support for SolidRun Clearfog GT 8K

sduponch opened the pull request:
Bump mvebu kernel to 4.19 (work from @ratkaj https://github.com/openwrt/openwrt/pull/1386)
Add SolidRun ClearFog-GT-8K platform support

sduponch committed:
mvebu: Add solidrun clearfog-gt-8k board support

ynezz commented on target/linux/mvebu/Makefile:
```diff
@@ -12,7 +12,7 @@ FEATURES:=fpu usb pci pcie gpio nand squashfs ramdisk
 MAINTAINER:=Imre Kaloz <kaloz@openwrt.org>
 
-KERNEL_PATCHVER:=4.14
+KERNEL_PATCHVER:=4.19
```
@sduponch I think, that it's too early for that.

adrianschmutzler replied:
If I remember correctly, kernel 4.19 may only become default after openwrt-19.01 has been created (which should be 4.14 for all targets).
So for now, I suppose this patch can be removed.

sduponch replied:
Sorry, doing...

ynezz commented:
@sduponch It doesn't build here

```
CONFIG_TARGET_mvebu=y
CONFIG_TARGET_mvebu_cortexa72=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_DEVICE_mvebu_cortexa72_DEVICE_armada-macchiatobin=y
CONFIG_TARGET_DEVICE_mvebu_cortexa72_DEVICE_armada-7040-db=y
CONFIG_TARGET_DEVICE_mvebu_cortexa72_DEVICE_armada-8040-db=y
CONFIG_TARGET_DEVICE_mvebu_cortexa72_DEVICE_solidrun-clearfog-gt-8k=y
CONFIG_ALL_KMODS=y
CONFIG_ALL_NONSHARED=y
CONFIG_TARGET_ALL_PROFILES=y
```

```
/home/petr/testing/openwrt/build_dir/target-aarch64_cortex-a72_musl/linux-mvebu_cortexa72/mwlwifi-2018-11-14-81413aa9/core.c: In function 'mwl_wl_init':
/home/petr/testing/openwrt/build_dir/target-aarch64_cortex-a72_musl/linux-mvebu_cortexa72/mwlwifi-2018-11-14-81413aa9/core.c:906:2: error: implicit declaration of function 'setup_timer'; did you mean 'sk_stop_timer'? [-Werror=implicit-function-declaration]
  setup_timer(&priv->period_timer, timer_routine, (unsigned long)hw);
  ^~~~~~~~~~~
  sk_stop_timer
 }
 ^
cc1: some warnings being treated as errors
scripts/Makefile.build:303: recipe for target '/home/petr/testing/openwrt/build_dir/target-aarch64_cortex-a72_musl/linux-mvebu_cortexa72/mwlwifi-2018-11-14-81413aa9/core.o' failed
```

sduponch commented:
Looking at it, thank you.

sduponch commented:
https://github.com/kaloz/mwlwifi/issues/338

dengqf6 commented:
`ledtrig-usbport` doesn't work

sduponch commented:
@ynezz the mwlwifi package is updated.

sduponch commented:
@dengqf6 which platform ?

sduponch commented:
How can i define a minimal kernel version for a target ? (ex: hide the clearfog-gt-8k while kernel is lower than 4.19)

dengqf6 commented:
>  which platform ?

Armada 385

nasbdh9 commented:
> > which platform ?
> 
> Armada 385

The OpenWRT tree in SolidRun has some patches for the Armada 385, I don't know if it works for 4.19.

https://github.com/SolidRun/openwrt/commits/openwrt-18.06-sr

ynezz commented:
@sduponch That was quick, but unfortunately there seems to be now kernel config problem:

```
net/sched/Kconfig:44: warning: menuconfig statement without prompt
.config:4069:warning: symbol value 'm' invalid for NF_NAT_REDIRECT
.config:5763:warning: override: SPARSEMEM_MANUAL changes choice state
*
* Restart config...
*
*
* Kernel Features
*
Page size
> 1. 4KB (ARM64_4K_PAGES)
  2. 16KB (ARM64_16K_PAGES)
  3. 64KB (ARM64_64K_PAGES)
...
ARM Scalable Vector Extension support (ARM64_SVE) [Y/n/?] (NEW)
```

You need to build in verbose mode to see this issue `make V=s`.

```
CONFIG_TARGET_mvebu=y
CONFIG_TARGET_mvebu_cortexa72=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_DEVICE_mvebu_cortexa72_DEVICE_armada-macchiatobin=y
CONFIG_TARGET_DEVICE_mvebu_cortexa72_DEVICE_armada-7040-db=y
CONFIG_TARGET_DEVICE_mvebu_cortexa72_DEVICE_armada-8040-db=y
CONFIG_TARGET_DEVICE_mvebu_cortexa72_DEVICE_solidrun-clearfog-gt-8k=y
CONFIG_ALL_KMODS=y
CONFIG_ALL_NONSHARED=y
CONFIG_TARGET_ALL_PROFILES=y
```

ynezz commented:
> ```
> net/sched/Kconfig:44: warning: menuconfig statement without prompt
> .config:4069:warning: symbol value 'm' invalid for NF_NAT_REDIRECT
> .config:5763:warning: override: SPARSEMEM_MANUAL changes choice state
> *
> * Restart config...
> *
> *
> * Kernel Features
> *
> Page size
> > 1. 4KB (ARM64_4K_PAGES)
>   2. 16KB (ARM64_16K_PAGES)
>   3. 64KB (ARM64_64K_PAGES)
> ...
> ARM Scalable Vector Extension support (ARM64_SVE) [Y/n/?] (NEW)
> ```

Same problem with following config:

```
CONFIG_TARGET_mvebu=y
CONFIG_TARGET_mvebu_cortexa53=y
CONFIG_TARGET_mvebu_cortexa53_Default=y
CONFIG_ALL_KMODS=y
CONFIG_ALL_NONSHARED=y
```

dengqf6 commented:
> Boot failure [log](http://termbin.com/tauh) on a wrt1900ac(V1 _mamba_). This is from a second attempt, the first booted but with a stream of _ubifs_ errors.

It seems that bad blocks are at critical area..

sduponch commented:
Thanks for your feedbacks, i will take a look ASAP.

nasbdh9 commented:
I am trying to compile a CF GT 8K image with this pr, then how do I set the uboot boot script?

sduponch commented:
@nasbdh9 I don't use the uboot boot script for now, i put the script only to match other platform behavior.

I use a custom uboot that i have forked from Solidrun : 
```
git clone --recurse-submodules  https://github.com/sduponch/clearfog-gt-8k-uboot.git
cd clearfog-gt-8k-uboot
./build.sh
sudo dd if=uboot.bin of=/dev/sdcard bs=512 seek=1
```
This will put the uboot on your sdcard and be able to boot, then you can write the uboot (from an USB key) directly to the SPI using "bubt uboot.bin spi usb".

dengqf6 commented on target/linux/mvebu/config-4.19:
```diff
@@ -0,0 +1,633 @@
+CONFIG_BLK_SCSI_REQUEST=y
+CONFIG_BOUNCE=y
+CONFIG_BPFILTER=y
+CONFIG_BPFILTER_UMH=y
```
Please do not enable bpfilter here, remove these 2 lines
I've observed network outage and an error message `bpfilter: read fail -512` in kernel log

sduponch replied:
Done

schm0 commented:
Seems to be working fine on wrt1200v1
Thank you!

kidome commented:
could anyone successfully booted share the new kernel size setting? wrt1900ac v2 seems not able to work with the old setting?

neheb commented:
Anyone know what patch 102-revert_i2c_delay.patch is for? It's been in the tree for ages. Since at least 3.18.

Some of these other patches also look questionable.

hauke commented on target/linux/mvebu/patches-4.19/423-ARM-dts-armada-385-linksys-Disable-internal-RTC.patch:
```diff
@@ -0,0 +1,28 @@
++
++&rtc {
++	/* No crystal connected to the internal RTC */
++	status = "disabled";
```
Please remove this patch, the change is already applied upstream

sduponch commented:
Thank you @hauke, i am trying to clean this up ASAP.

hauke commented:
When I build this for the cortexa72 target I am missing these kernel configuration options:
```
CONFIG_ARM64_SVE
CONFIG_USB_ROLE_SWITCH
```

ynezz commented:
@hauke CONFIG_USB_ROLE_SWITCH is queued in https://git.openwrt.org/?p=openwrt/staging/chunkeey.git;a=blobdiff;f=target/linux/generic/config-4.19;h=8809fbc08370134e1774fe85faab542dd65bb57d;hp=a3470e955cb8011bf835e5e9a2648d39ef0c18f1;hb=b0d9db7ef9b9b1fa4ea33f27d0a4c0907ec1396f;hpb=18e4492389ae4607da59c9fe3492f0db998efb5f

sduponch commented:
> > ```
> > net/sched/Kconfig:44: warning: menuconfig statement without prompt
> > .config:4069:warning: symbol value 'm' invalid for NF_NAT_REDIRECT
> > .config:5763:warning: override: SPARSEMEM_MANUAL changes choice state
> > *
> > * Restart config...
> > *
> > *
> > * Kernel Features
> > *
> > Page size
> > > 1. 4KB (ARM64_4K_PAGES)
> >   2. 16KB (ARM64_16K_PAGES)
> >   3. 64KB (ARM64_64K_PAGES)
> > ...
> > ARM Scalable Vector Extension support (ARM64_SVE) [Y/n/?] (NEW)
> > ```
> 
> Same problem with following config:
> 
> ```
> CONFIG_TARGET_mvebu=y
> CONFIG_TARGET_mvebu_cortexa53=y
> CONFIG_TARGET_mvebu_cortexa53_Default=y
> CONFIG_ALL_KMODS=y
> CONFIG_ALL_NONSHARED=y
> ```

These symbols looks seted by config-default in target directory : 
~/openwrt/target/linux/mvebu$ grep -r ARM64_4K_PAGES
cortexa53/config-default:CONFIG_ARM64_4K_PAGES=y
cortexa72/config-default:CONFIG_ARM64_4K_PAGES=y

sduponch commented:
Sorry i was not available lately : 
- Rebased
- removed 423-ARM-dts-armada-385-linksys-Disable-internal-RTC.patch
- Added CONFIG_ARM64_SVE in A53/A72 config-default

neheb commented:
That RTC patch is needed (and is a backport) as the hardware has no crystal.

sduponch commented:
> That RTC patch is needed (and is a backport) as the hardware has no crystal.

Patch look already applied on linux 4.19.25 : 

```
~/openwrt/build_dir~/openwrt/build_dir/target-aarch64_cortex-a72_musl/linux-mvebu_cortexa72/linux-4.19.25$ grep rtc --after-context 3 ./arch/arm/boot/dts/armada-385-linksys.dtsi
&rtc {
	/* No crystal connected to the internal RTC */
	status = "disabled";
};
```

sduponch commented:
```
.config:4124:warning: symbol value 'm' invalid for NF_NAT_REDIRECT
.config:5852:warning: override: SPARSEMEM_MANUAL changes choice state
*
* Restart config...
*
*
* Kernel Features
*
Page size
> 1. 4KB (ARM64_4K_PAGES)
  2. 16KB (ARM64_16K_PAGES)
  3. 64KB (ARM64_64K_PAGES)
choice[1-3?]: 1
Virtual address space size
> 1. 39-bit (ARM64_VA_BITS_39)
  2. 48-bit (ARM64_VA_BITS_48)
choice[1-2?]: 1
Physical address space size
> 1. 48-bit (ARM64_PA_BITS_48) (NEW)
choice[1]: 1
Build big-endian kernel (CPU_BIG_ENDIAN) [N/y/?] n
Multi-core scheduler support (SCHED_MC) [N/y/?] n
SMT scheduler support (SCHED_SMT) [N/y/?] n
Maximum number of CPUs (2-4096) (NR_CPUS) [4] 4
Support for hot-pluggable CPUs (HOTPLUG_CPU) [Y/n/?] y
Numa Memory Allocation and Scheduler Support (NUMA) [N/y/?] n
Timer frequency
> 1. 100 HZ (HZ_100)
  2. 250 HZ (HZ_250)
  3. 300 HZ (HZ_300)
  4. 1000 HZ (HZ_1000)
choice[1-4?]: 1
Enable seccomp to safely compute untrusted bytecode (SECCOMP) [N/y/?] n
Enable paravirtualization code (PARAVIRT) [N/y/?] n
Paravirtual steal time accounting (PARAVIRT_TIME_ACCOUNTING) [N/y/?] n
Build kdump crash kernel (CRASH_DUMP) [Y/n/?] y
Xen guest support on ARM64 (XEN) [N/y/?] n
Unmap kernel when running in userspace (aka "KAISER") (UNMAP_KERNEL_AT_EL0) [Y/n/?] y
Harden the branch predictor against aliasing attacks (HARDEN_BRANCH_PREDICTOR) [Y/n/?] y
Harden EL2 vector mapping against system register leak (HARDEN_EL2_VECTORS) [Y/n/?] y
Speculative Store Bypass Disable (ARM64_SSBD) [Y/n/?] y
Emulate Privileged Access Never using TTBR0_EL1 switching (ARM64_SW_TTBR0_PAN) [N/y/?] n
ARM Scalable Vector Extension support (ARM64_SVE) [Y/n/?] y
Randomize the address of the kernel image (RANDOMIZE_BASE) [N/y/?] n
```
Build is displaying this at build but does not prompt for values.

ynezz commented:
> Build is displaying this at build but does not prompt for values.

@sduponch Just run `make kernel_oldconfig` to fix those missing kernel symbols as well. This is how it looks here after `make kernel_oldconfig`:

```
diff --git a/target/linux/mvebu/config-4.19 b/target/linux/mvebu/config-4.19
index 275f897980..4a4628c89b 100644
--- a/target/linux/mvebu/config-4.19
+++ b/target/linux/mvebu/config-4.19
@@ -1,114 +1,127 @@
+CONFIG_64BIT=y
 CONFIG_AHCI_MVEBU=y
-CONFIG_ALIGNMENT_TRAP=y
 CONFIG_ARCH_CLOCKSOURCE_DATA=y
+CONFIG_ARCH_DMA_ADDR_T_64BIT=y
+CONFIG_ARCH_HAS_CACHE_LINE_SIZE=y
 CONFIG_ARCH_HAS_DEBUG_VIRTUAL=y
 CONFIG_ARCH_HAS_ELF_RANDOMIZE=y
+CONFIG_ARCH_HAS_FAST_MULTIPLIER=y
 CONFIG_ARCH_HAS_FORTIFY_SOURCE=y
 CONFIG_ARCH_HAS_GCOV_PROFILE_ALL=y
 CONFIG_ARCH_HAS_KCOV=y
 CONFIG_ARCH_HAS_MEMBARRIER_SYNC_CORE=y
-CONFIG_ARCH_HAS_PHYS_TO_DMA=y
+CONFIG_ARCH_HAS_PTE_SPECIAL=y
 CONFIG_ARCH_HAS_SET_MEMORY=y
 CONFIG_ARCH_HAS_SG_CHAIN=y
 CONFIG_ARCH_HAS_STRICT_KERNEL_RWX=y
 CONFIG_ARCH_HAS_STRICT_MODULE_RWX=y
+CONFIG_ARCH_HAS_SYSCALL_WRAPPER=y
 CONFIG_ARCH_HAS_TICK_BROADCAST=y
-CONFIG_ARCH_HAVE_CUSTOM_GPIO_H=y
+CONFIG_ARCH_HAVE_NMI_SAFE_CMPXCHG=y
 CONFIG_ARCH_HIBERNATION_POSSIBLE=y
-CONFIG_ARCH_MIGHT_HAVE_PC_PARPORT=y
-CONFIG_ARCH_MULTIPLATFORM=y
-CONFIG_ARCH_MULTI_V6_V7=y
-CONFIG_ARCH_MULTI_V7=y
+CONFIG_ARCH_INLINE_READ_LOCK=y
+CONFIG_ARCH_INLINE_READ_LOCK_BH=y
+CONFIG_ARCH_INLINE_READ_LOCK_IRQ=y
+CONFIG_ARCH_INLINE_READ_LOCK_IRQSAVE=y
+CONFIG_ARCH_INLINE_READ_UNLOCK=y
+CONFIG_ARCH_INLINE_READ_UNLOCK_BH=y
+CONFIG_ARCH_INLINE_READ_UNLOCK_IRQ=y
+CONFIG_ARCH_INLINE_READ_UNLOCK_IRQRESTORE=y
+CONFIG_ARCH_INLINE_SPIN_LOCK=y
+CONFIG_ARCH_INLINE_SPIN_LOCK_BH=y
+CONFIG_ARCH_INLINE_SPIN_LOCK_IRQ=y
+CONFIG_ARCH_INLINE_SPIN_LOCK_IRQSAVE=y
+CONFIG_ARCH_INLINE_SPIN_TRYLOCK=y
+CONFIG_ARCH_INLINE_SPIN_TRYLOCK_BH=y
+CONFIG_ARCH_INLINE_SPIN_UNLOCK=y
+CONFIG_ARCH_INLINE_SPIN_UNLOCK_BH=y
+CONFIG_ARCH_INLINE_SPIN_UNLOCK_IRQ=y
+CONFIG_ARCH_INLINE_SPIN_UNLOCK_IRQRESTORE=y
+CONFIG_ARCH_INLINE_WRITE_LOCK=y
+CONFIG_ARCH_INLINE_WRITE_LOCK_BH=y
+CONFIG_ARCH_INLINE_WRITE_LOCK_IRQ=y
+CONFIG_ARCH_INLINE_WRITE_LOCK_IRQSAVE=y
+CONFIG_ARCH_INLINE_WRITE_UNLOCK=y
+CONFIG_ARCH_INLINE_WRITE_UNLOCK_BH=y
+CONFIG_ARCH_INLINE_WRITE_UNLOCK_IRQ=y
+CONFIG_ARCH_INLINE_WRITE_UNLOCK_IRQRESTORE=y
+CONFIG_ARCH_MMAP_RND_BITS=18
+CONFIG_ARCH_MMAP_RND_BITS_MAX=24
+CONFIG_ARCH_MMAP_RND_BITS_MIN=18
+CONFIG_ARCH_MMAP_RND_COMPAT_BITS_MIN=11
 CONFIG_ARCH_MVEBU=y
-# CONFIG_ARCH_NPCM is not set
-CONFIG_ARCH_NR_GPIO=0
-CONFIG_ARCH_OPTIONAL_KERNEL_RWX=y
-CONFIG_ARCH_OPTIONAL_KERNEL_RWX_DEFAULT=y
-# CONFIG_ARCH_STM32 is not set
+CONFIG_ARCH_PROC_KCORE_TEXT=y
+CONFIG_ARCH_SELECT_MEMORY_MODEL=y
+CONFIG_ARCH_SPARSEMEM_DEFAULT=y
+CONFIG_ARCH_SPARSEMEM_ENABLE=y
 CONFIG_ARCH_SUPPORTS_ATOMIC_RMW=y
-CONFIG_ARCH_SUPPORTS_BIG_ENDIAN=y
+CONFIG_ARCH_SUPPORTS_DEBUG_PAGEALLOC=y
+CONFIG_ARCH_SUPPORTS_INT128=y
+CONFIG_ARCH_SUPPORTS_MEMORY_FAILURE=y
+CONFIG_ARCH_SUPPORTS_NUMA_BALANCING=y
 CONFIG_ARCH_SUPPORTS_UPROBES=y
 CONFIG_ARCH_SUSPEND_POSSIBLE=y
-CONFIG_ARCH_USE_BUILTIN_BSWAP=y
 CONFIG_ARCH_USE_CMPXCHG_LOCKREF=y
-CONFIG_ARCH_WANT_GENERAL_HUGETLB=y
-CONFIG_ARCH_WANT_IPC_PARSE_VERSION=y
-CONFIG_ARM=y
-CONFIG_ARMADA375_USBCLUSTER_PHY=y
-CONFIG_ARMADA_370_CLK=y
-CONFIG_ARMADA_370_XP_IRQ=y
-CONFIG_ARMADA_370_XP_TIMER=y
-CONFIG_ARMADA_375_CLK=y
-CONFIG_ARMADA_38X_CLK=y
-CONFIG_ARMADA_39X_CLK=y
+CONFIG_ARCH_USE_QUEUED_RWLOCKS=y
+CONFIG_ARCH_USE_QUEUED_SPINLOCKS=y
+CONFIG_ARCH_WANT_COMPAT_IPC_PARSE_VERSION=y
+CONFIG_ARCH_WANT_FRAME_POINTERS=y
+CONFIG_ARCH_WANT_HUGE_PMD_SHARE=y
+CONFIG_ARM64=y
+# CONFIG_ARM64_16K_PAGES is not set
+CONFIG_ARM64_4K_PAGES=y
+# CONFIG_ARM64_64K_PAGES is not set
+CONFIG_ARM64_CONT_SHIFT=4
+# CONFIG_ARM64_CRYPTO is not set
+CONFIG_ARM64_HW_AFDBM=y
+CONFIG_ARM64_LSE_ATOMICS=y
+CONFIG_ARM64_PAGE_SHIFT=12
+CONFIG_ARM64_PAN=y
+CONFIG_ARM64_PA_BITS=48
+CONFIG_ARM64_PA_BITS_48=y
+# CONFIG_ARM64_PMEM is not set
+# CONFIG_ARM64_PTDUMP_DEBUGFS is not set
+# CONFIG_ARM64_RANDOMIZE_TEXT_OFFSET is not set
+CONFIG_ARM64_SSBD=y
+CONFIG_ARM64_SVE=y
+# CONFIG_ARM64_SW_TTBR0_PAN is not set
+CONFIG_ARM64_UAO=y
+CONFIG_ARM64_VA_BITS=39
+CONFIG_ARM64_VA_BITS_39=y
+# CONFIG_ARM64_VA_BITS_48 is not set
+CONFIG_ARM64_VHE=y
+CONFIG_ARMADA_37XX_CLK=y
+CONFIG_ARMADA_AP806_SYSCON=y
+CONFIG_ARMADA_CP110_SYSCON=y
 CONFIG_ARMADA_THERMAL=y
-CONFIG_ARMADA_XP_CLK=y
-CONFIG_ARM_APPENDED_DTB=y
+CONFIG_ARM_AMBA=y
+CONFIG_ARM_ARCH_TIMER=y
+CONFIG_ARM_ARCH_TIMER_EVTSTREAM=y
 # CONFIG_ARM_ARMADA_37XX_CPUFREQ is not set
-CONFIG_ARM_ATAG_DTB_COMPAT=y
-# CONFIG_ARM_ATAG_DTB_COMPAT_CMDLINE_EXTEND is not set
-# CONFIG_ARM_ATAG_DTB_COMPAT_CMDLINE_FROM_BOOTLOADER is not set
-CONFIG_ARM_ATAG_DTB_COMPAT_CMDLINE_MANGLE=y
-CONFIG_ARM_CPU_SUSPEND=y
-CONFIG_ARM_CRYPTO=y
-CONFIG_ARM_ERRATA_720789=y
-CONFIG_ARM_ERRATA_764369=y
 CONFIG_ARM_GIC=y
-CONFIG_ARM_GLOBAL_TIMER=y
-CONFIG_ARM_HAS_SG_CHAIN=y
-CONFIG_ARM_HEAVY_MB=y
-CONFIG_ARM_L1_CACHE_SHIFT=6
-CONFIG_ARM_L1_CACHE_SHIFT_6=y
-# CONFIG_ARM_LPAE is not set
-CONFIG_ARM_MVEBU_V7_CPUIDLE=y
-CONFIG_ARM_PATCH_IDIV=y
-CONFIG_ARM_PATCH_PHYS_VIRT=y
-# CONFIG_ARM_PTDUMP_DEBUGFS is not set
-CONFIG_ARM_THUMB=y
-# CONFIG_ARM_THUMBEE is not set
-CONFIG_ARM_UNWIND=y
-CONFIG_ARM_VIRT_EXT=y
-# CONFIG_ASIX_PHY is not set
+CONFIG_ARM_GIC_V2M=y
+CONFIG_ARM_GIC_V3=y
+CONFIG_ARM_GIC_V3_ITS=y
+CONFIG_ARM_GIC_V3_ITS_PCI=y
+# CONFIG_ARM_PL172_MPMC is not set
+CONFIG_ARM_PSCI_FW=y
+# CONFIG_ARM_SP805_WATCHDOG is not set
 CONFIG_ASYNC_TX_ENABLE_CHANNEL_SWITCH=y
 CONFIG_ATA=y
-CONFIG_ATAGS=y
-# CONFIG_AUTOFS_FS is not set
-CONFIG_AUTO_ZRELADDR=y
-# CONFIG_BINFMT_ELF_FDPIC is not set
+CONFIG_AUDIT_ARCH_COMPAT_GENERIC=y
 CONFIG_BLK_DEV_LOOP=y
 CONFIG_BLK_DEV_SD=y
 CONFIG_BLK_MQ_PCI=y
 CONFIG_BLK_SCSI_REQUEST=y
-CONFIG_BOUNCE=y
-CONFIG_BUILD_SALT=""
-# CONFIG_CACHE_FEROCEON_L2 is not set
-CONFIG_CACHE_L2X0=y
-CONFIG_CC_HAS_SANCOV_TRACE_PC=y
-CONFIG_CC_HAS_STACKPROTECTOR_NONE=y
-CONFIG_CC_IS_GCC=y
 # CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE is not set
 CONFIG_CC_OPTIMIZE_FOR_SIZE=y
-CONFIG_CLANG_VERSION=0
 CONFIG_CLKDEV_LOOKUP=y
-CONFIG_CLKSRC_ARM_GLOBAL_TIMER_SCHED_CLOCK=y
-CONFIG_CLKSRC_MMIO=y
 CONFIG_CLONE_BACKWARDS=y
 CONFIG_COMMON_CLK=y
-# CONFIG_COMMON_CLK_MAX9485 is not set
-# CONFIG_COMMON_CLK_SI544 is not set
-CONFIG_CONSOLE_LOGLEVEL_QUIET=4
+# CONFIG_COMPAT is not set
 CONFIG_CPUFREQ_DT=y
 CONFIG_CPUFREQ_DT_PLATDEV=y
-CONFIG_CPU_32v6K=y
-CONFIG_CPU_32v7=y
-CONFIG_CPU_ABRT_EV7=y
 # CONFIG_CPU_BIG_ENDIAN is not set
-# CONFIG_CPU_BPREDICT_DISABLE is not set
-CONFIG_CPU_CACHE_V7=y
-CONFIG_CPU_CACHE_VIPT=y
-CONFIG_CPU_COPY_V6=y
-CONFIG_CPU_CP15=y
-CONFIG_CPU_CP15_MMU=y
 CONFIG_CPU_FREQ=y
 CONFIG_CPU_FREQ_DEFAULT_GOV_ONDEMAND=y
 # CONFIG_CPU_FREQ_DEFAULT_GOV_PERFORMANCE is not set
@@ -120,96 +133,43 @@ CONFIG_CPU_FREQ_GOV_PERFORMANCE=y
 # CONFIG_CPU_FREQ_GOV_POWERSAVE is not set
 # CONFIG_CPU_FREQ_GOV_USERSPACE is not set
 CONFIG_CPU_FREQ_STAT=y
-CONFIG_CPU_HAS_ASID=y
 # CONFIG_CPU_HOTPLUG_STATE_CONTROL is not set
-# CONFIG_CPU_ICACHE_DISABLE is not set
 CONFIG_CPU_IDLE=y
 CONFIG_CPU_IDLE_GOV_LADDER=y
 CONFIG_CPU_ISOLATION=y
-CONFIG_CPU_PABRT_V7=y
-CONFIG_CPU_PJ4B=y
 CONFIG_CPU_PM=y
 CONFIG_CPU_RMAP=y
-CONFIG_CPU_SPECTRE=y
 CONFIG_CPU_THERMAL=y
-CONFIG_CPU_THUMB_CAPABLE=y
-CONFIG_CPU_TLB_V7=y
-CONFIG_CPU_V7=y
 CONFIG_CRASH_DUMP=y
 CONFIG_CRC16=y
-# CONFIG_CRC64 is not set
 CONFIG_CRYPTO_ACOMP2=y
 CONFIG_CRYPTO_AEAD=y
 CONFIG_CRYPTO_AEAD2=y
-# CONFIG_CRYPTO_AEGIS128 is not set
-# CONFIG_CRYPTO_AEGIS128L is not set
-# CONFIG_CRYPTO_AEGIS256 is not set
-CONFIG_CRYPTO_AES_ARM=y
-CONFIG_CRYPTO_AES_ARM_BS=y
-# CONFIG_CRYPTO_AES_ARM_CE is not set
-# CONFIG_CRYPTO_CFB is not set
-# CONFIG_CRYPTO_CHACHA20_NEON is not set
 CONFIG_CRYPTO_CRC32=y
 CONFIG_CRYPTO_CRC32C=y
-# CONFIG_CRYPTO_CRC32_ARM_CE is not set
 CONFIG_CRYPTO_CRYPTD=y
 CONFIG_CRYPTO_DEFLATE=y
 CONFIG_CRYPTO_DES=y
 CONFIG_CRYPTO_DEV_MARVELL_CESA=y
-# CONFIG_CRYPTO_GHASH_ARM_CE is not set
 CONFIG_CRYPTO_HASH=y
 CONFIG_CRYPTO_HASH2=y
 CONFIG_CRYPTO_HW=y
 CONFIG_CRYPTO_LZO=y
 CONFIG_CRYPTO_MANAGER=y
 CONFIG_CRYPTO_MANAGER2=y
-# CONFIG_CRYPTO_MORUS1280 is not set
-# CONFIG_CRYPTO_MORUS640 is not set
 CONFIG_CRYPTO_NULL2=y
 CONFIG_CRYPTO_RNG2=y
 CONFIG_CRYPTO_SHA1=y
-CONFIG_CRYPTO_SHA1_ARM=y
-# CONFIG_CRYPTO_SHA1_ARM_CE is not set
-CONFIG_CRYPTO_SHA1_ARM_NEON=y
-CONFIG_CRYPTO_SHA256_ARM=y
-# CONFIG_CRYPTO_SHA2_ARM_CE is not set
-CONFIG_CRYPTO_SHA512_ARM=y
-CONFIG_CRYPTO_SIMD=y
-# CONFIG_CRYPTO_SM3 is not set
-# CONFIG_CRYPTO_SM4 is not set
-# CONFIG_CRYPTO_SPECK is not set
-# CONFIG_CRYPTO_SPECK_NEON is not set
 CONFIG_CRYPTO_WORKQUEUE=y
-# CONFIG_CRYPTO_ZSTD is not set
 CONFIG_DCACHE_WORD_ACCESS=y
 CONFIG_DEBUG_ALIGN_RODATA=y
 CONFIG_DEBUG_INFO=y
-CONFIG_DEBUG_LL=y
-CONFIG_DEBUG_LL_INCLUDE="debug/8250.S"
-CONFIG_DEBUG_MVEBU_UART0=y
-# CONFIG_DEBUG_MVEBU_UART0_ALTERNATE is not set
-# CONFIG_DEBUG_MVEBU_UART1_ALTERNATE is not set
-# CONFIG_DEBUG_RSEQ is not set
-# CONFIG_DEBUG_RWSEMS is not set
-CONFIG_DEBUG_UART_8250=y
-# CONFIG_DEBUG_UART_8250_FLOW_CONTROL is not set
-CONFIG_DEBUG_UART_8250_SHIFT=2
-# CONFIG_DEBUG_UART_8250_WORD is not set
-CONFIG_DEBUG_UART_PHYS=0xd0012000
-CONFIG_DEBUG_UART_VIRT=0xfec12000
-CONFIG_DEBUG_UNCOMPRESS=y
-CONFIG_DEBUG_USER=y
 CONFIG_DMADEVICES=y
+CONFIG_DMA_DIRECT_OPS=y
 CONFIG_DMA_ENGINE=y
 CONFIG_DMA_ENGINE_RAID=y
 CONFIG_DMA_OF=y
-# CONFIG_DP83822_PHY is not set
-# CONFIG_DP83TC811_PHY is not set
-# CONFIG_DRM_DP_CEC is not set
 CONFIG_DTC=y
-# CONFIG_DW_AXI_DMAC is not set
-CONFIG_EARLY_PRINTK=y
-CONFIG_EDAC_ATOMIC_SCRUB=y
 CONFIG_EDAC_SUPPORT=y
 CONFIG_EXT4_FS=y
 CONFIG_EXTCON=y
@@ -218,24 +178,24 @@ CONFIG_F2FS_FS=y
 # CONFIG_F2FS_FS_SECURITY is not set
 CONFIG_F2FS_FS_XATTR=y
 CONFIG_F2FS_STAT_FS=y
-# CONFIG_FAILOVER is not set
-# CONFIG_FIND_BIT_BENCHMARK is not set
 CONFIG_FIXED_PHY=y
 CONFIG_FIX_EARLYCON_MEM=y
+CONFIG_FRAME_POINTER=y
 CONFIG_FS_IOMAP=y
 CONFIG_FS_MBCACHE=y
-# CONFIG_FTWDT010_WATCHDOG is not set
-CONFIG_GCC_VERSION=70300
 CONFIG_GENERIC_ALLOCATOR=y
 CONFIG_GENERIC_ARCH_TOPOLOGY=y
 CONFIG_GENERIC_BUG=y
+CONFIG_GENERIC_BUG_RELATIVE_POINTERS=y
 CONFIG_GENERIC_CLOCKEVENTS=y
 CONFIG_GENERIC_CLOCKEVENTS_BROADCAST=y
 CONFIG_GENERIC_CPU_AUTOPROBE=y
+CONFIG_GENERIC_CSUM=y
 CONFIG_GENERIC_EARLY_IOREMAP=y
 CONFIG_GENERIC_IDLE_POLL_SETUP=y
 CONFIG_GENERIC_IRQ_CHIP=y
 CONFIG_GENERIC_IRQ_EFFECTIVE_AFF_MASK=y
+CONFIG_GENERIC_IRQ_MIGRATION=y
 CONFIG_GENERIC_IRQ_MULTI_HANDLER=y
 CONFIG_GENERIC_IRQ_SHOW=y
 CONFIG_GENERIC_IRQ_SHOW_LEVEL=y
@@ -243,91 +203,105 @@ CONFIG_GENERIC_MSI_IRQ=y
 CONFIG_GENERIC_MSI_IRQ_DOMAIN=y
 CONFIG_GENERIC_PCI_IOMAP=y
 CONFIG_GENERIC_PHY=y
+CONFIG_GENERIC_PINCONF=y
 CONFIG_GENERIC_SCHED_CLOCK=y
 CONFIG_GENERIC_SMP_IDLE_THREAD=y
 CONFIG_GENERIC_STRNCPY_FROM_USER=y
 CONFIG_GENERIC_STRNLEN_USER=y
+CONFIG_GENERIC_TIME_VSYSCALL=y
 CONFIG_GLOB=y
-# CONFIG_GNSS is not set
 CONFIG_GPIOLIB=y
-CONFIG_GPIOLIB_FASTPATH_LIMIT=512
 CONFIG_GPIOLIB_IRQCHIP=y
 CONFIG_GPIO_GENERIC=y
 CONFIG_GPIO_GENERIC_PLATFORM=y
-# CONFIG_GPIO_HLWD is not set
-# CONFIG_GPIO_MAX3191X is not set
-# CONFIG_GPIO_MB86S7X is not set
 CONFIG_GPIO_MVEBU=y
 CONFIG_GPIO_PCA953X=y
 CONFIG_GPIO_PCA953X_IRQ=y
-# CONFIG_GPIO_PCIE_IDIO_24 is not set
 CONFIG_GPIO_SYSFS=y
-# CONFIG_GUP_BENCHMARK is not set
 CONFIG_HANDLE_DOMAIN_IRQ=y
 CONFIG_HARDEN_BRANCH_PREDICTOR=y
 CONFIG_HARDIRQS_SW_RESEND=y
 CONFIG_HAS_DMA=y
 CONFIG_HAS_IOMEM=y
 CONFIG_HAS_IOPORT_MAP=y
+CONFIG_HAVE_ALIGNED_STRUCT_PAGE=y
 CONFIG_HAVE_ARCH_AUDITSYSCALL=y
 CONFIG_HAVE_ARCH_BITREVERSE=y
+CONFIG_HAVE_ARCH_HUGE_VMAP=y
 CONFIG_HAVE_ARCH_JUMP_LABEL=y
+CONFIG_HAVE_ARCH_KASAN=y
 CONFIG_HAVE_ARCH_KGDB=y
 CONFIG_HAVE_ARCH_PFN_VALID=y
+CONFIG_HAVE_ARCH_PREL32_RELOCATIONS=y
 CONFIG_HAVE_ARCH_SECCOMP_FILTER=y
 CONFIG_HAVE_ARCH_THREAD_STRUCT_WHITELIST=y
 CONFIG_HAVE_ARCH_TRACEHOOK=y
-CONFIG_HAVE_ARM_SCU=y
+CONFIG_HAVE_ARCH_TRANSPARENT_HUGEPAGE=y
+CONFIG_HAVE_ARCH_VMAP_STACK=y
 CONFIG_HAVE_ARM_SMCCC=y
-CONFIG_HAVE_ARM_TWD=y
 CONFIG_HAVE_CLK=y
 CONFIG_HAVE_CLK_PREPARE=y
+CONFIG_HAVE_CMPXCHG_DOUBLE=y
+CONFIG_HAVE_CMPXCHG_LOCAL=y
 CONFIG_HAVE_CONTEXT_TRACKING=y
 CONFIG_HAVE_C_RECORDMCOUNT=y
+CONFIG_HAVE_DEBUG_BUGVERBOSE=y
 CONFIG_HAVE_DEBUG_KMEMLEAK=y
 CONFIG_HAVE_DMA_CONTIGUOUS=y
 CONFIG_HAVE_DYNAMIC_FTRACE=y
-CONFIG_HAVE_DYNAMIC_FTRACE_WITH_REGS=y
 CONFIG_HAVE_EBPF_JIT=y
 CONFIG_HAVE_EFFICIENT_UNALIGNED_ACCESS=y
 CONFIG_HAVE_FTRACE_MCOUNT_RECORD=y
 CONFIG_HAVE_FUNCTION_GRAPH_TRACER=y
 CONFIG_HAVE_FUNCTION_TRACER=y
 CONFIG_HAVE_GENERIC_DMA_COHERENT=y
-CONFIG_HAVE_IDE=y
+CONFIG_HAVE_GENERIC_GUP=y
 CONFIG_HAVE_IRQ_TIME_ACCOUNTING=y
 CONFIG_HAVE_MEMBLOCK=y
-CONFIG_HAVE_MOD_ARCH_SPECIFIC=y
 CONFIG_HAVE_NET_DSA=y
-CONFIG_HAVE_OPROFILE=y
-CONFIG_HAVE_OPTPROBES=y
+CONFIG_HAVE_PATA_PLATFORM=y
 CONFIG_HAVE_PERF_EVENTS=y
 CONFIG_HAVE_PERF_REGS=y
 CONFIG_HAVE_PERF_USER_STACK_DUMP=y
-CONFIG_HAVE_PROC_CPU=y
+CONFIG_HAVE_RCU_TABLE_FREE=y
 CONFIG_HAVE_REGS_AND_STACK_ACCESS_API=y
 CONFIG_HAVE_RSEQ=y
-CONFIG_HAVE_SMP=y
-CONFIG_HAVE_STACKPROTECTOR=y
 CONFIG_HAVE_SYSCALL_TRACEPOINTS=y
-CONFIG_HAVE_UID16=y
 CONFIG_HAVE_VIRT_CPU_ACCOUNTING_GEN=y
-CONFIG_HIGHMEM=y
-# CONFIG_HIGHPTE is not set
+CONFIG_HOLES_IN_ZONE=y
 CONFIG_HOTPLUG_CPU=y
-CONFIG_HWBM=y
+# CONFIG_HUGETLBFS is not set
 CONFIG_HWMON=y
 CONFIG_HW_RANDOM=y
 CONFIG_HW_RANDOM_OMAP=y
-CONFIG_HZ_FIXED=0
 CONFIG_HZ_PERIODIC=y
 CONFIG_I2C=y
 CONFIG_I2C_BOARDINFO=y
 CONFIG_I2C_CHARDEV=y
 CONFIG_I2C_MV64XXX=y
 # CONFIG_I2C_PXA is not set
-# CONFIG_ICE is not set
+CONFIG_ILLEGAL_POINTER_VALUE=0xdead000000000000
 CONFIG_INITRAMFS_SOURCE=""
+CONFIG_INLINE_READ_LOCK=y
+CONFIG_INLINE_READ_LOCK_BH=y
+CONFIG_INLINE_READ_LOCK_IRQ=y
+CONFIG_INLINE_READ_LOCK_IRQSAVE=y
+CONFIG_INLINE_READ_UNLOCK_BH=y
+CONFIG_INLINE_READ_UNLOCK_IRQRESTORE=y
+CONFIG_INLINE_SPIN_LOCK=y
+CONFIG_INLINE_SPIN_LOCK_BH=y
+CONFIG_INLINE_SPIN_LOCK_IRQ=y
+CONFIG_INLINE_SPIN_LOCK_IRQSAVE=y
+CONFIG_INLINE_SPIN_TRYLOCK=y
+CONFIG_INLINE_SPIN_TRYLOCK_BH=y
+CONFIG_INLINE_SPIN_UNLOCK_BH=y
+CONFIG_INLINE_SPIN_UNLOCK_IRQRESTORE=y
+CONFIG_INLINE_WRITE_LOCK=y
+CONFIG_INLINE_WRITE_LOCK_BH=y
+CONFIG_INLINE_WRITE_LOCK_IRQ=y
+CONFIG_INLINE_WRITE_LOCK_IRQSAVE=y
+CONFIG_INLINE_WRITE_UNLOCK_BH=y
+CONFIG_INLINE_WRITE_UNLOCK_IRQRESTORE=y
 CONFIG_IP6_NF_IPTABLES=y
 CONFIG_IP6_NF_MATCH_SRH=y
 CONFIG_IPV6=y
@@ -336,66 +310,39 @@ CONFIG_IRQ_DOMAIN=y
 CONFIG_IRQ_DOMAIN_HIERARCHY=y
 CONFIG_IRQ_FORCED_THREADING=y
 CONFIG_IRQ_WORK=y
-# CONFIG_IWMMXT is not set
 CONFIG_JBD2=y
-# CONFIG_LAN743X is not set
-# CONFIG_LEDS_CR0014114 is not set
 CONFIG_LEDS_GPIO=y
-# CONFIG_LEDS_LM3692X is not set
-# CONFIG_LEDS_MLXREG is not set
 CONFIG_LEDS_PCA963X=y
 CONFIG_LEDS_TLC591XX=y
-# CONFIG_LEDS_TRIGGER_ACTIVITY is not set
 CONFIG_LEDS_TRIGGER_DISK=y
 CONFIG_LIBFDT=y
 CONFIG_LOCK_DEBUGGING_SUPPORT=y
 CONFIG_LOCK_SPIN_ON_OWNER=y
 CONFIG_LZO_COMPRESS=y
 CONFIG_LZO_DECOMPRESS=y
-CONFIG_MACH_ARMADA_370=y
-CONFIG_MACH_ARMADA_375=y
-CONFIG_MACH_ARMADA_38X=y
-CONFIG_MACH_ARMADA_39X=y
-CONFIG_MACH_ARMADA_XP=y
-# CONFIG_MACH_DOVE is not set
-CONFIG_MACH_MVEBU_ANY=y
-CONFIG_MACH_MVEBU_V7=y
 CONFIG_MAGIC_SYSRQ=y
 CONFIG_MANGLE_BOOTARGS=y
 CONFIG_MARVELL_PHY=y
 CONFIG_MDIO_BUS=y
 CONFIG_MDIO_DEVICE=y
 CONFIG_MDIO_I2C=y
-# CONFIG_MDIO_MSCC_MIIM is not set
-# CONFIG_MELLANOX_PLATFORM is not set
 CONFIG_MEMFD_CREATE=y
 CONFIG_MEMORY=y
-# CONFIG_MFD_MADERA is not set
-# CONFIG_MFD_ROHM_BD718XX is not set
 CONFIG_MFD_SYSCON=y
-# CONFIG_MICROCHIP_T1_PHY is not set
-CONFIG_MIGHT_HAVE_CACHE_L2X0=y
-CONFIG_MIGHT_HAVE_PCI=y
-# CONFIG_MISC_RTSX_PCI is not set
-# CONFIG_MISC_RTSX_USB is not set
+CONFIG_MIGRATION=y
 CONFIG_MMC=y
 CONFIG_MMC_BLOCK=y
-# CONFIG_MMC_CQHCI is not set
-CONFIG_MMC_MVSDIO=y
 CONFIG_MMC_SDHCI=y
-# CONFIG_MMC_SDHCI_OF_DWCMSHC is not set
-# CONFIG_MMC_SDHCI_OMAP is not set
 # CONFIG_MMC_SDHCI_PCI is not set
 CONFIG_MMC_SDHCI_PLTFM=y
 CONFIG_MMC_SDHCI_PXAV3=y
 # CONFIG_MMC_TIFM_SD is not set
-CONFIG_MODULES_USE_ELF_REL=y
+CONFIG_MODULES_USE_ELF_RELA=y
 CONFIG_MTD_CFI_STAA=y
 CONFIG_MTD_M25P80=y
 CONFIG_MTD_NAND=y
 CONFIG_MTD_NAND_ECC=y
 CONFIG_MTD_NAND_MARVELL=y
-# CONFIG_MTD_SPI_NAND is not set
 CONFIG_MTD_SPI_NOR=y
 CONFIG_MTD_SPLIT_FIRMWARE=y
 CONFIG_MTD_UBI=y
@@ -404,26 +351,20 @@ CONFIG_MTD_UBI_BLOCK=y
 # CONFIG_MTD_UBI_FASTMAP is not set
 # CONFIG_MTD_UBI_GLUEBI is not set
 CONFIG_MTD_UBI_WL_THRESHOLD=4096
-# CONFIG_MTK_MMC is not set
 CONFIG_MUTEX_SPIN_ON_OWNER=y
-CONFIG_MVEBU_CLK_COMMON=y
-CONFIG_MVEBU_CLK_COREDIV=y
-CONFIG_MVEBU_CLK_CPU=y
-CONFIG_MVEBU_DEVBUS=y
-CONFIG_MVEBU_MBUS=y
+CONFIG_MVEBU_GICP=y
+CONFIG_MVEBU_ICU=y
+CONFIG_MVEBU_ODMI=y
+CONFIG_MVEBU_PIC=y
 CONFIG_MVMDIO=y
 CONFIG_MVNETA=y
-CONFIG_MVNETA_BM=y
-CONFIG_MVNETA_BM_ENABLE=y
 CONFIG_MVPP2=y
 CONFIG_MVSW61XX_PHY=y
 CONFIG_MV_XOR=y
 CONFIG_NEED_DMA_MAP_STATE=y
-CONFIG_NEON=y
-# CONFIG_NETDEVSIM is not set
+CONFIG_NEED_SG_DMA_LENGTH=y
 CONFIG_NETFILTER=y
 CONFIG_NETFILTER_ADVANCED=y
-# CONFIG_NETFILTER_NETLINK_OSF is not set
 CONFIG_NETFILTER_XTABLES=y
 CONFIG_NET_DSA=y
 CONFIG_NET_DSA_LEGACY=y
@@ -434,27 +375,20 @@ CONFIG_NET_DSA_MV88E6XXX_GLOBAL2=y
 CONFIG_NET_DSA_TAG_DSA=y
 CONFIG_NET_DSA_TAG_EDSA=y
 # CONFIG_NET_DSA_VITESSE_VSC73XX is not set
-# CONFIG_NET_FAILOVER is not set
 CONFIG_NET_FLOW_LIMIT=y
-# CONFIG_NET_SCH_CAKE is not set
-# CONFIG_NET_SCH_CBS is not set
-# CONFIG_NET_SCH_ETF is not set
-# CONFIG_NET_SCH_SKBPRIO is not set
 CONFIG_NET_SWITCHDEV=y
-CONFIG_NET_VENDOR_CADENCE=y
 # CONFIG_NET_VENDOR_CORTINA is not set
 # CONFIG_NET_VENDOR_MICROSEMI is not set
-CONFIG_NET_VENDOR_NETERION=y
 # CONFIG_NET_VENDOR_NI is not set
-CONFIG_NET_VENDOR_PACKET_ENGINES=y
 # CONFIG_NET_VENDOR_SOCIONEXT is not set
 # CONFIG_NF_LOG_IPV6 is not set
 CONFIG_NF_TPROXY_IPV4=y
-# CONFIG_NF_TPROXY_IPV6 is not set
 CONFIG_NLS=y
 CONFIG_NOP_USB_XCEIV=y
 CONFIG_NO_BOOTMEM=y
 CONFIG_NR_CPUS=4
+# CONFIG_NUMA is not set
+CONFIG_NVMEM=y
 CONFIG_OF=y
 CONFIG_OF_ADDRESS=y
 CONFIG_OF_EARLY_FLATTREE=y
@@ -465,53 +399,37 @@ CONFIG_OF_KOBJ=y
 CONFIG_OF_MDIO=y
 CONFIG_OF_NET=y
 CONFIG_OF_RESERVED_MEM=y
-CONFIG_OLD_SIGACTION=y
-CONFIG_OLD_SIGSUSPEND3=y
-CONFIG_ORION_WATCHDOG=y
-CONFIG_OUTER_CACHE=y
-CONFIG_OUTER_CACHE_SYNC=y
-# CONFIG_OVERLAY_FS_METACOPY is not set
-CONFIG_OVERLAY_FS_REDIRECT_ALWAYS_FOLLOW=y
 # CONFIG_OVERLAY_FS_XINO_AUTO is not set
 CONFIG_PADATA=y
-CONFIG_PAGE_OFFSET=0xC0000000
+CONFIG_PARTITION_PERCPU=y
 CONFIG_PCI=y
-# CONFIG_PCIE_CADENCE_HOST is not set
-# CONFIG_PCIE_DW_PLAT_HOST is not set
-# CONFIG_PCIE_XILINX is not set
+# CONFIG_PCI_AARDVARK is not set
 CONFIG_PCI_DOMAINS=y
 CONFIG_PCI_DOMAINS_GENERIC=y
 CONFIG_PCI_MSI=y
 CONFIG_PCI_MSI_IRQ_DOMAIN=y
-CONFIG_PCI_MVEBU=y
-# CONFIG_PCI_V3_SEMI is not set
-CONFIG_PERF_USE_VMALLOC=y
-CONFIG_PGTABLE_LEVELS=2
+CONFIG_PGTABLE_LEVELS=3
 CONFIG_PHYLIB=y
 CONFIG_PHYLINK=y
-# CONFIG_PHY_MAPPHONE_MDM6600 is not set
+CONFIG_PHYS_ADDR_T_64BIT=y
 # CONFIG_PHY_MVEBU_CP110_COMPHY is not set
 CONFIG_PINCTRL=y
-CONFIG_PINCTRL_ARMADA_370=y
-CONFIG_PINCTRL_ARMADA_375=y
-CONFIG_PINCTRL_ARMADA_38X=y
-CONFIG_PINCTRL_ARMADA_39X=y
-CONFIG_PINCTRL_ARMADA_XP=y
+CONFIG_PINCTRL_ARMADA_37XX=y
+CONFIG_PINCTRL_ARMADA_AP806=y
+CONFIG_PINCTRL_ARMADA_CP110=y
 CONFIG_PINCTRL_MVEBU=y
 # CONFIG_PINCTRL_SINGLE is not set
-CONFIG_PJ4B_ERRATA_4742=y
-# CONFIG_PL310_ERRATA_588369 is not set
-# CONFIG_PL310_ERRATA_727915 is not set
-CONFIG_PL310_ERRATA_753970=y
-# CONFIG_PL310_ERRATA_769419 is not set
-CONFIG_PLAT_ORION=y
-CONFIG_PLUGIN_HOSTCC="g++"
 CONFIG_PM_OPP=y
+CONFIG_POWER_RESET=y
+CONFIG_POWER_SUPPLY=y
 CONFIG_PRINTK_SAFE_LOG_BUF_SHIFT=11
 CONFIG_PROC_VMCORE=y
 CONFIG_PROC_VMCORE_DEVICE_DUMP=y
 CONFIG_PWM=y
 CONFIG_PWM_SYSFS=y
+CONFIG_QUEUED_RWLOCKS=y
+CONFIG_QUEUED_SPINLOCKS=y
+# CONFIG_RANDOMIZE_BASE is not set
 CONFIG_RATIONAL=y
 CONFIG_RCU_NEED_SEGCBLIST=y
 CONFIG_RCU_STALL_COMMON=y
@@ -521,46 +439,32 @@ CONFIG_REGMAP_I2C=y
 CONFIG_REGMAP_MMIO=y
 CONFIG_REGMAP_SPI=y
 CONFIG_REGULATOR=y
-# CONFIG_REGULATOR_88PG86X is not set
 CONFIG_REGULATOR_FIXED_VOLTAGE=y
-# CONFIG_REGULATOR_SY8106A is not set
-# CONFIG_RENESAS_PHY is not set
 CONFIG_RFS_ACCEL=y
-# CONFIG_RPMSG_VIRTIO is not set
 CONFIG_RPS=y
 CONFIG_RSEQ=y
 CONFIG_RTC_CLASS=y
 CONFIG_RTC_DRV_ARMADA38X=y
-# CONFIG_RTC_DRV_ISL12026 is not set
 CONFIG_RTC_DRV_MV=y
-# CONFIG_RTC_DRV_PCF85363 is not set
 CONFIG_RTC_I2C_AND_SPI=y
-CONFIG_RTC_MC146818_LIB=y
-CONFIG_RUNTIME_TESTING_MENU=y
 CONFIG_RWSEM_SPIN_ON_OWNER=y
 CONFIG_RWSEM_XCHGADD_ALGORITHM=y
 CONFIG_SATA_AHCI_PLATFORM=y
 CONFIG_SATA_MV=y
 CONFIG_SATA_PMP=y
 CONFIG_SCSI=y
-# CONFIG_SENSORS_MAX6621 is not set
-# CONFIG_SENSORS_NPCM7XX is not set
 CONFIG_SENSORS_PWM_FAN=y
 CONFIG_SENSORS_TMP421=y
-# CONFIG_SENSORS_W83773G is not set
 CONFIG_SERIAL_8250_DW=y
 CONFIG_SERIAL_8250_FSL=y
+# CONFIG_SERIAL_AMBA_PL011 is not set
 CONFIG_SERIAL_MVEBU_CONSOLE=y
 CONFIG_SERIAL_MVEBU_UART=y
 CONFIG_SFP=y
 CONFIG_SGL_ALLOC=y
 CONFIG_SG_POOL=y
-# CONFIG_SIOX is not set
-# CONFIG_SLIMBUS is not set
 CONFIG_SMP=y
-CONFIG_SMP_ON_UP=y
-CONFIG_SOC_BUS=y
-# CONFIG_SOUNDWIRE is not set
+CONFIG_SPARSEMEM_VMEMMAP_ENABLE=y
 CONFIG_SPARSE_IRQ=y
 CONFIG_SPI=y
 # CONFIG_SPI_ARMADA_3700 is not set
@@ -568,38 +472,32 @@ CONFIG_SPI_MASTER=y
 CONFIG_SPI_MEM=y
 CONFIG_SPI_ORION=y
 CONFIG_SRAM=y
-CONFIG_SRAM_EXEC=y
 CONFIG_SRCU=y
 CONFIG_STACKPROTECTOR=y
 CONFIG_STACKPROTECTOR_STRONG=y
 CONFIG_SWCONFIG=y
+CONFIG_SWIOTLB=y
 CONFIG_SWPHY=y
-CONFIG_SWP_EMULATE=y
-CONFIG_SYS_SUPPORTS_APM_EMULATION=y
-# CONFIG_TEST_BITFIELD is not set
-# CONFIG_TEST_IDA is not set
-# CONFIG_TEST_OVERFLOW is not set
+CONFIG_SYSCTL_EXCEPTION_TRACE=y
+CONFIG_SYS_SUPPORTS_HUGETLBFS=y
 CONFIG_THERMAL=y
 CONFIG_THERMAL_DEFAULT_GOV_STEP_WISE=y
 CONFIG_THERMAL_EMERGENCY_POWEROFF_DELAY_MS=0
 CONFIG_THERMAL_GOV_STEP_WISE=y
 CONFIG_THERMAL_HWMON=y
 CONFIG_THERMAL_OF=y
-# CONFIG_THERMAL_STATISTICS is not set
-# CONFIG_THUMB2_KERNEL is not set
+CONFIG_THREAD_INFO_IN_TASK=y
 CONFIG_TICK_CPU_ACCOUNTING=y
 CONFIG_TIMER_OF=y
 CONFIG_TIMER_PROBE=y
 CONFIG_TREE_RCU=y
 CONFIG_TREE_SRCU=y
-# CONFIG_TYPEC is not set
 CONFIG_UBIFS_FS=y
 # CONFIG_UBIFS_FS_ADVANCED_COMPR is not set
 CONFIG_UBIFS_FS_LZO=y
 CONFIG_UBIFS_FS_XATTR=y
 CONFIG_UBIFS_FS_ZLIB=y
-CONFIG_UNCOMPRESS_INCLUDE="debug/uncompress.h"
-# CONFIG_UNISYSSPAR is not set
+CONFIG_UNMAP_KERNEL_AT_EL0=y
 CONFIG_USB=y
 CONFIG_USB_COMMON=y
 CONFIG_USB_EHCI_HCD=y
@@ -609,23 +507,15 @@ CONFIG_USB_LEDS_TRIGGER_USBPORT=y
 CONFIG_USB_PHY=y
 CONFIG_USB_STORAGE=y
 CONFIG_USB_SUPPORT=y
-# CONFIG_USB_XHCI_DBGCAP is not set
 CONFIG_USB_XHCI_HCD=y
 CONFIG_USB_XHCI_MVEBU=y
 CONFIG_USB_XHCI_PLATFORM=y
-CONFIG_USE_OF=y
-CONFIG_VFP=y
-CONFIG_VFPv3=y
-CONFIG_VIRTIO_MENU=y
+CONFIG_VMAP_STACK=y
 CONFIG_WATCHDOG_CORE=y
 CONFIG_XDP_SOCKETS=y
-# CONFIG_XILINX_VCU is not set
-# CONFIG_XIL_AXIS_FIFO is not set
 CONFIG_XPS=y
 CONFIG_XZ_DEC_ARM=y
 CONFIG_XZ_DEC_BCJ=y
-CONFIG_ZBOOT_ROM_BSS=0x0
-CONFIG_ZBOOT_ROM_TEXT=0x0
 CONFIG_ZLIB_DEFLATE=y
 CONFIG_ZLIB_INFLATE=y
-# CONFIG_NF_CONNTRACK_LABELS is not set
+CONFIG_ZONE_DMA32=y
```

sduponch commented:
This mixup ARMv7 and ARM64 (cortex A53/A72) in the same config-default...

ynezz commented:
> This mixup ARMv7 and ARM64 (cortex A53/A72) in the same config-default...

Sorry, my bad `make kernel_oldconfig CONFIG_TARGET=subtarget`

sduponch commented:
No problem, it looks much better, tank you :)

sduponch commented:
- run oldconfig on mvebu/config-4.19
- run oldconfig on cortexa72/config-default

sduponch commented:
- run oldconfig on cortexa72/config-default

sduponch commented:
> Meanwhile I had time to build another profile with 4.19
> 
> ```
> CONFIG_TARGET_mvebu=y
> CONFIG_TARGET_mvebu_cortexa9=y
> CONFIG_TARGET_MULTI_PROFILE=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_globalscale-mirabox=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_linksys-wrt1200ac=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_linksys-wrt1900ac=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_linksys-wrt1900acs=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_linksys-wrt1900acv2=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_linksys-wrt3200acm=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_linksys-wrt32x=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-370-db=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-370-rd=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-385-db-ap=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-388-rd=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-xp-db=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-xp-gp=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_openblocks-ax3-4=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-388-clearfog-base=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-388-clearfog-pro=y
> CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_turris-omnia=y
> CONFIG_ALL_KMODS=y
> CONFIG_ALL_NONSHARED=y
> CONFIG_TARGET_ALL_PROFILES=y
> ```
> 
> and it ends with following error:
> 
> ```
> Configuring kmod-gpio-Collected errors:
>  * satisfy_dependencies_for: Cannot satisfy the following dependencies for kmod-bluetooth_6lowpan:
>  *      kmod-6lowpan * 
>  * opkg_install_cmd: Cannot install package kmod-bluetooth_6lowpan.
> ```

This issue does not look specific to mvebu...
6lowpan is well listed as dependency of bluetooth_6lowpan in  package/kernel/linux/modules/other.mk.

```
define KernelPackage/bluetooth_6lowpan
  SUBMENU:=$(OTHER_MENU)
  TITLE:=Bluetooth 6LoWPAN support
  DEPENDS:=+kmod-6lowpan +kmod-bluetooth
  KCONFIG:=CONFIG_BT_6LOWPAN
  FILES:=$(LINUX_DIR)/net/bluetooth/bluetooth_6lowpan.ko
  AUTOLOAD:=$(call AutoProbe,bluetooth_6lowpan)
endef
```
Maybe 6lowpan is available only on 64bits platform.

sduponch commented:
https://cateee.net/lkddb/web-lkddb/6LOWPAN.html
6lowpan depends on IPV6 that is not defined in : 
```
define KernelPackage/6lowpan
  SUBMENU:=$(OTHER_MENU)
  TITLE:=6LoWPAN shared code
  KCONFIG:= \
»···CONFIG_6LOWPAN \
»···CONFIG_6LOWPAN_NHC=n
  FILES:=$(LINUX_DIR)/net/6lowpan/6lowpan.ko
  AUTOLOAD:=$(call AutoProbe,6lowpan)
endef
```

ynezz commented:
@sduponch Thanks, I've just now successfully finished build with following config:

```
CONFIG_TARGET_mvebu=y
CONFIG_TARGET_mvebu_cortexa72=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_DEVICE_mvebu_cortexa72_DEVICE_armada-macchiatobin=y
CONFIG_TARGET_DEVICE_mvebu_cortexa72_DEVICE_armada-7040-db=y
CONFIG_TARGET_DEVICE_mvebu_cortexa72_DEVICE_armada-8040-db=y
CONFIG_TARGET_DEVICE_mvebu_cortexa72_DEVICE_solidrun-clearfog-gt-8k=y
CONFIG_ALL_KMODS=y
CONFIG_ALL_NONSHARED=y
CONFIG_TARGET_ALL_PROFILES=y
```

ynezz commented:
> This issue does not look specific to mvebu...
6lowpan is well listed as dependency of bluetooth_6lowpan in package/kernel/linux/modules/other.mk.

It seems like it's somehow specific to mvebu, as I can't reproduce it for example on ramips with manually selected kmod-bluetooth_6lowpan package. But this issue is not something you're eligible to fix. Maybe @jow- has an idea?

Anyway, I was able to "fix it" by adding `CONFIG_TARGET_PER_DEVICE_ROOTFS=y` into my .config, so I was able to build following config succesfully as well:

```
CONFIG_TARGET_mvebu=y
CONFIG_TARGET_mvebu_cortexa9=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_globalscale-mirabox=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_globalscale-mirabox=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_linksys-wrt1200ac=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_linksys-wrt1200ac=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_linksys-wrt1900ac=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_linksys-wrt1900ac=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_linksys-wrt1900acs=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_linksys-wrt1900acs=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_linksys-wrt1900acv2=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_linksys-wrt1900acv2=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_linksys-wrt3200acm=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_linksys-wrt3200acm=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_linksys-wrt32x=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_linksys-wrt32x=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-370-db=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_armada-370-db=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-370-rd=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_armada-370-rd=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-385-db-ap=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_armada-385-db-ap=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-388-rd=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_armada-388-rd=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-xp-db=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_armada-xp-db=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-xp-gp=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_armada-xp-gp=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_openblocks-ax3-4=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_openblocks-ax3-4=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-388-clearfog-base=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_armada-388-clearfog-base=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_armada-388-clearfog-pro=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_armada-388-clearfog-pro=""
CONFIG_TARGET_DEVICE_mvebu_cortexa9_DEVICE_turris-omnia=y
CONFIG_TARGET_DEVICE_PACKAGES_mvebu_cortexa9_DEVICE_turris-omnia=""
CONFIG_TARGET_PER_DEVICE_ROOTFS=y
CONFIG_TARGET_ALL_PROFILES=y
```

Good work, thanks!

anomeome commented:
I see message:
`[    0.945000] pxa3xx-nand f10d0000.nand: This platform can't do DMA on this device`
in [4.14 boot log](https://termbin.com/mana) of a _mamba_ device, but not in [4.19 boot log](https://termbin.com/jw3v) with subsequent failure on _nand_ access followed by boot failure.

Are there any reports of successful deployment on a _mamba_?

sduponch commented:
> I see message:
> `[ 0.945000] pxa3xx-nand f10d0000.nand: This platform can't do DMA on this device`
> in [4.14 boot log](https://termbin.com/mana) of a _mamba_ device, but not in [4.19 boot log](https://termbin.com/jw3v) with subsequent failure on _nand_ access followed by boot failure.
> 
> Are there any reports of successful deployment on a _mamba_?

Maybe a missing module or a bad dtsi file, if you solve the conf to get it work please send me the fix.

anomeome commented:
The 4.14 CONFIG_MTD_NAND_PXA3xx has been replaced in 4.19 with the new CONFIG_MTD_NAND_MARVELL [driver](https://patchwork.kernel.org/patch/10123271/), ~14 months back so I assume it has had time be be shaken out by now. If I uncover any new information I will update.

sduponch commented:
> The 4.14 CONFIG_MTD_NAND_PXA3xx has been replaced in 4.19 with the new CONFIG_MTD_NAND_MARVELL [driver](https://patchwork.kernel.org/patch/10123271/), ~14 months back so I assume it has had time be be shaken out by now. If I uncover any new information I will update.



> The 4.14 CONFIG_MTD_NAND_PXA3xx has been replaced in 4.19 with the new CONFIG_MTD_NAND_MARVELL [driver](https://patchwork.kernel.org/patch/10123271/), ~14 months back so I assume it has had time be be shaken out by now. If I uncover any new information I will update.
```

[    0.989817] nand: device found, Manufacturer ID: 0x01, Chip ID: 0xf1
[    0.996215] nand: AMD/Spansion S34ML01G1
[    1.000150] nand: 128 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
[    1.007987] Bad block table found at page 65472, version 0x02
[    1.014120] Bad block table found at page 65408, version 0x03
[    1.020063] nand_read_bbt: bad block at 0x0000023e0000
[    1.025227] nand_read_bbt: bad block at 0x0000029e0000
[    1.030395] nand_read_bbt: bad block at 0x000007f60000
[    1.035546] nand_read_bbt: bad block at 0x000007f80000
[    1.040705] nand_read_bbt: bad block at 0x000007fa0000
[    1.045856] nand_read_bbt: bad block at 0x000007fe0000
[    1.051045] nand: nand_erase_nand: attempt to erase a bad block at page 0x0000ffc0
[    1.058634] nand_bbt: error while erasing BBT block -5
[    1.063838] nand_bbt: error -30 while marking block 1023 bad
[    1.069536] nand_bbt: error while erasing BBT block -5
[    1.075434] nand_bbt: error while erasing BBT block -5
[    1.080651] nand_bbt: error -30 while marking block 1017 bad
[    1.086350] nand_bbt: error while erasing BBT block -5
[    1.091564] nand_bbt: error -30 while marking block 1016 bad
[    1.097239] No space left to write bad block table
[    1.102056] nand_bbt: error while writing bad block table -28
[    1.107822] marvell-nfc f10d0000.nand-controller: could not scan the nand chip
[    1.115108] marvell-nfc: probe of f10d0000.nand-controller failed with error -28
```

hauke commented:
When compiling this with kernel 4.14 for the espressobin I am getting a question for a kernel config option, I would suggest to split the config-default into two configs like config-4.14 and config-4.19.

hauke commented on target/linux/mvebu/cortexa72/config-default:
```diff
@@ -75,39 +110,57 @@ CONFIG_HAVE_GENERIC_GUP=y
 CONFIG_HAVE_RCU_TABLE_FREE=y
+CONFIG_HOLES_IN_ZONE=y
 # CONFIG_HUGETLBFS is not set
-CONFIG_I2C_MUX_PCA954x=y
```
This should not be removed.

sduponch replied:
Done

anomeome commented:
[Forum dialogue](https://forum.openwrt.org/t/kernel-4-19-upgrade-issue-on-mvebu-target-s/32966)

diizzyy commented:
Since this seems like a bit of a mess it's probably easier to start with a vanilla stock config and diff it compared to the one that gets generated by buildroot and work from there. That's more or less what I did with the Octeon target which was a lot easier.

nasbdh9 commented:
any progress?
 I am using sduponch uboot to try to start gt8k but did not succeed, did not recognize my sd card or usb key
@sduponch

dengqf6 commented:
@sduponch plz rebase the PR

nasbdh9 commented:
My gt8k can now start openwrt correctly, but 88E6141 can't work normally after connecting wan port, without any error, any help?

tmn505 commented:
Hi @sduponch 
You are adding a lot of additional config symbols, this should be added in another commit with explanation why these are needed.
I rebased the work of @ratkaj on current state of master (https://github.com/tmn505/openwrt/tree/mvebu-4.19), and would like to send it to ML, is that OK with You?

@anomeome 
can You post again Your failed mamba dmesg, it seems like it expired. BTW maybe this patch `104-linksys_mamba_disable_keep_config.patch` is not needed? This commit message https://git.kernel.org/torvalds/c/02f26ecf8c772751d4b24744d487f6b1b20e75d4, explains that new driver fixes few previous bugs.

nasbdh9 commented:
After testing at https://github.com/tmn505/openwrt/tree/mvebu-4.19, I found that this commit (https://github.com/tmn505/openwrt/commit/9e3a8332830ab5d298e4f22df27612d18aa7751a) will cause the device to Kernel panic, log: https://gist.github.com/nasbdh9/3ce4cf161646924ecf1f4eacc7e8af75 @tmn505 

There are some fault warnings when starting GT8K with this configuration in this commit (https://github.com/openwrt/openwrt/pull/1646/commits/d4fc4217f944919794db0d5d5c8b290d0981ea57), after which the switch will not work. log: https://gist.github.com/nasbdh9/6f298b5128b1d3fd5242e9be29936904 @sduponch

tmn505 commented:
> After testing at https://github.com/tmn505/openwrt/tree/mvebu-4.19, I found that this commit (tmn505@9e3a833) will cause the device to Kernel panic, log: https://gist.github.com/nasbdh9/3ce4cf161646924ecf1f4eacc7e8af75

I don't see any kernel panic in Your kernel log, just missing firmware. Anyway, I moved the crypto driver to package, since the firmware for EIP197 variants is not freely available (has to be obtained by Marvell after signing NDA). Now You won't see this error if this module won't be enabled.

sduponch commented:
> Hi @sduponch
> You are adding a lot of additional config symbols, this should be added in another commit with explanation why these are needed.
> I rebased the work of @ratkaj on current state of master (https://github.com/tmn505/openwrt/tree/mvebu-4.19), and would like to send it to ML, is that OK with You?

@tmn505 Ok for me.

> There are some fault warnings when starting GT8K with this configuration in this commit (d4fc421), 
> after which the switch will not work. log: https://gist.github.com/nasbdh9/6f298b5128b1d3fd5242e9be29936904 @sduponch

@nasbdh9 i need to update my build but everything was working fine, i will take a look ASAP. Until now i was building a custom uboot but GT8K support in uboot should be in mainline now.

tmn505 commented:
> @tmn505 Ok for me.

Thanks, Marek (@elkablo) was faster: http://lists.infradead.org/pipermail/openwrt-devel/2019-May/017145.html. I agreed to his proposal for merging both our work. I'm expecting v2 soonish.

v-vid commented:
Hi @tmn505,

I've tested your rebased patch on ClearFog Base with additional BiWin M2 SSD, Champion 1000SFPT module and mPCIE wireless card and can confirm that everything above works as well as the standard features. 

Do you have any ETA on when this could be merged?  I'm currently working on adding ATF package for mvebu but it depends on the 4.19 kernel.

tmn505 commented:
> I've tested your rebased patch on ClearFog Base with additional BiWin M2 SSD, Champion 1000SFPT module and mPCIE wireless card and can confirm that everything above works as well as the standard features.

Thanks.

> Do you have any ETA on when this could be merged? I'm currently working on adding ATF package for mvebu but it depends on the 4.19 kernel.

Don't know yet when Marek will submit second version on the ML (merged mine and his changes). But after that it should be straight forward (around one to three weeks). I'll ask him about the progress.

tmn505 commented:
FYI, anyone wanting to test kernel bump: https://patchwork.ozlabs.org/patch/1110513. Anyones `Tested-by:` would be appreciated.

jglooije commented:
> FYI, anyone wanting to test kernel bump: https://patchwork.ozlabs.org/patch/1110513. Anyones `Tested-by:` would be appreciated.

Tested the patch on WRT1900ACS no problems after one day of running no strang messages in the logs

EXSERENS commented:
> FYI, anyone wanting to test kernel bump: https://patchwork.ozlabs.org/patch/1110513. Anyones `Tested-by:` would be appreciated.

running without problem on WRT3200ACM with GCC 9.1.0 -O3 for all day long.

ynezz commented:
> FYI, anyone wanting to test kernel bump: https://patchwork.ozlabs.org/patch/1110513. Anyones `Tested-by:` would be appreciated.

4.19 has landed in the tree, so in order to continue, this should be rebased

tmn505 commented on target/linux/mvebu/image/solidrun-clearfog-gt-8k.bootscript:
```diff
@@ -0,0 +1,10 @@
+setenv bootargs "root=PARTUUID=@ROOT@-02 rw rootwait"
```
This boot script can be removed if You only want to support booting from MMC, there is already generic one provided with `Default-arm64`. But as I can see from images provided by SolidRun https://images.solid-build.xyz/8040/, default bootloader also provides distro boot variables (https://gitlab.denx.de/u-boot/u-boot/raw/master/doc/README.distro) so booting from USB should be possible with altering `load` commands a bit, which examples I'll show below.

tmn505 commented on target/linux/mvebu/image/solidrun-clearfog-gt-8k.bootscript:
```diff
@@ -0,0 +1,10 @@
+	setenv bootargs "${bootargs} ${console}"
+fi
+
+load mmc 1:1 ${fdt_addr_r} armada-8040-clearfog-gt-8k.dtb
```
e.g. : `load ${devtype} ${devnum}:${partition} ${fdt_addr_r} @DTB@.dtb`

tmn505 commented on target/linux/mvebu/image/solidrun-clearfog-gt-8k.bootscript:
```diff
@@ -0,0 +1,10 @@
+fi
+
+load mmc 1:1 ${fdt_addr_r} armada-8040-clearfog-gt-8k.dtb
+load mmc 1:1 ${kernel_addr_r} Image
```
e.g. : `load ${devtype} ${devnum}:${partition} ${kernel_addr_r} Image`

tmn505 commented on target/linux/mvebu/patches-4.19/301-PCI-armada8k-add-support-for-gpio-controlled-reset-signal.patch:
```diff
@@ -0,0 +1,48 @@
+diff --git a/drivers/pci/controller/dwc/pcie-armada8k.c b/drivers/pci/controller/dwc/pcie-armada8k.c
```
This should be backport of upstream commit with proper message and don't forget to refresh patch with `make target/linux/{clean,refresh} V=s`.
On the other side, this patch got later reverted https://git.kernel.org/torvalds/c/f14bcc0add3abecceca1a3fe538c4ec9566893f3 for clashing with MACCHIATObin, is this patch needed for PCIe ports to function? And does this commit https://gitlab.denx.de/u-boot/u-boot/commit/d7f165cf676f4ae8aafd7adfc6dbe0f851026e46 make this patch obsolete? If yes, I would urge SolidRun to issue updated u-boot images on their site and mention in commit message that the bootloader needs re-flashing for PCIe ports to function.

tmn505 commented on target/linux/mvebu/patches-4.19/302-Armada-8040-clearfog-gt-8k.dts.patch:
```diff
@@ -0,0 +1,475 @@
+diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
```
This should be a series of backported patches with proper messages from Linus tree or dts file dropped in `target/linux/mvebu/files-4.19/arch/arm64/boot/dts/marvell`.

tmn505 reviewed (changes requested):
This PR still lacks proper commit message with a bit information about the hardware, flashing instructions (if applicable), what doesn't work (if applicable) and SoB line, consult https://openwrt.org/submitting-patches for broader explanation.
For last I have mixed feelings about renaming network interfaces but that's accepted by core team, then let it be.
Good luck.
